parts/k8s/kuberneteswinagentresourcesvmss.t

```diff
@@ -0,0 +1,193 @@
+{{if .IsStorageAccount}}
```

Comment:
what is the diff of this file with kubernetesagentresourcesvmss.t?

Reply:
74c74
<       "resourceNameSuffix" : "[variables('winResourceNamePrefix')]",
---
>       "resourceNameSuffix" : "[variables('nameSuffix')]",
131,134c131,148
<           "computerNamePrefix": "[concat(substring(variables('nameSuffix'), 0, 5), 'acs')]",
<           {{GetKubernetesWindowsAgentCustomData .}}
<           "adminUsername": "[variables('windowsAdminUsername')]",
<           "adminPassword": "[variables('windowsAdminPassword')]"
---
>           "adminUsername": "[variables('username')]",
>           "computerNamePrefix": "[variables('{{.Name}}VMNamePrefix')]",
>           {{GetKubernetesAgentCustomData .}}
>           "linuxConfiguration": {
>               "disablePasswordAuthentication": "true",
>               "ssh": {
>                 "publicKeys": [
>                   {
>                     "keyData": "[parameters('sshRSAPublicKey')]",
>                     "path": "[variables('sshKeyPath')]"
>                   }
>                 ]
>               }
>             }
>             {{if HasLinuxSecrets}}
>               ,
>               "secrets": "[variables('linuxProfileSecrets')]"
>             {{end}}
137c151,153
<           {{GetDataDisks .}}
---
>           {{if not (UseAgentCustomImage .)}}
>             {{GetDataDisks .}}
>           {{end}}
139,142c155,162
<             "offer": "[variables('agentWindowsOffer')]",
<             "publisher": "[variables('agentWindowsPublisher')]",
<             "sku": "[variables('agentWindowsSku')]",
<             "version": "[variables('agentWindowsVersion')]"
---
>             {{if UseAgentCustomImage .}}
>             "id": "[resourceId(variables('{{.Name}}osImageResourceGroup'), 'Microsoft.Compute/images', variables('{{.Name}}osImageName'))]"
>             {{else}}
>             "offer": "[variables('{{.Name}}osImageOffer')]",
>             "publisher": "[variables('{{.Name}}osImagePublisher')]",
>             "sku": "[variables('{{.Name}}osImageSKU')]",
>             "version": "[variables('{{.Name}}osImageVersion')]"
>             {{end}}
163,165c183,185
<                 "publisher": "Microsoft.Compute",
<                 "type": "CustomScriptExtension",
<                 "typeHandlerVersion": "1.8",
---
>                 "publisher": "Microsoft.Azure.Extensions",
>                 "type": "CustomScript",
>                 "typeHandlerVersion": "2.0",
169c189
<                     "commandToExecute": "[concat('powershell.exe -ExecutionPolicy Unrestricted -command \"', '$arguments = ', variables('singleQuote'),'-MasterIP ',variables('kubernetesAPIServerIP'),' -KubeDnsServiceIp ',variables('kubeDnsServiceIp'),' -MasterFQDNPrefix ',variables('masterFqdnPrefix'),' -Location ',variables('location'),' -AgentKey ',variables('clientPrivateKey'),' -AADClientId ',variables('servicePrincipalClientId'),' -AADClientSecret ',variables('servicePrincipalClientSecret'),variables('singleQuote'), ' ; ', variables('windowsCustomScriptSuffix'), '\" > %SYSTEMDRIVE%\\AzureData\\CustomDataSetupScript.log 2>&1')]"
---
>                   "commandToExecute": "[concat(variables('provisionScriptParametersCommon'),' /usr/bin/nohup /bin/bash -c \"/bin/bash /opt/azure/containers/provision.sh >> /var/log/azure/cluster-provision.log 2>&1\"')]"
178c198
<                 "type": "ManagedIdentityExtensionForWindows",
---
>                 "type": "ManagedIdentityExtensionForLinux",